ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -63,31 +67,44 @@ governor_BOOTSTRAP_PEERS_TIMEOUT = 15 * 60
 -- state) then, until the node reaches a clean state, this monitoring action
 -- will be disabled and thus churning will be disabled as well.
 --
+-- On the other hand, if Genesis mode is on for the node, this action responds
+-- to changes in ledger state judgement monitoring actions to change the static
+-- set of target peers.
 targetPeers :: (MonadSTM m, Ord peeraddr)
             => PeerSelectionActions peeraddr peerconn m
             -> PeerSelectionState peeraddr peerconn
             -> Guarded (STM m) (TimedDecision m peeraddr peerconn)
-targetPeers PeerSelectionActions{readPeerSelectionTargets}
+targetPeers PeerSelectionActions{ readPeerSelectionTargets,
+                                  readLedgerStateJudgement,
+                                  peerTargets = ConsensusModePeerTargets { praosTargets, genesisSyncTargets },
+                                  churnMutex }
             st@PeerSelectionState{
               publicRootPeers,
               localRootPeers,
               targets,
-              ledgerStateJudgement,
               bootstrapPeersFlag,
-              hasOnlyBootstrapPeers
+              hasOnlyBootstrapPeers,
+              consensusMode
             } =
     Guarded Nothing $ do
-      targets' <- readPeerSelectionTargets
-      check ( isNodeAbleToMakeProgress bootstrapPeersFlag
-                                       ledgerStateJudgement
-                                       hasOnlyBootstrapPeers
-            && targets' /= targets
-            && sanePeerSelectionTargets targets'
-            )
-      -- We simply ignore target updates that are not "sane".
-
+      targets'  <- readPeerSelectionTargets
+      ledgerStateJudgement' <- readLedgerStateJudgement
```

Comment:
why do we read `ledgerStateJudgement` here? This value should be considered up to date, it is the first action that gets checked, if there's a race does it really matter?

Reply:
I suppose you're correct - I've wrote this part a while ago but my understanding of the mechanics here has evolved since then :)

Reply:
Done